api/pkg/delete/delete.go

```diff
@@ -0,0 +1,83 @@
+package delete
```

Comment:
I understand the intention behind the separate package for `delete` operations (since the `client/mlflow` package simply implements the Go client for the Mlflow APIs). But I'm not sure we need this distinction. Also, `api/pkg/delete` doesn't seem like a reusable package as it's too broad.

Can this delete functionality simply be merged with the Mlflow package? I think it's ok to let the mlflow delete experiment method directly call the `gcs` package util. And in this case, it may be better to simply use `mlflow` as the package in place of `client/mlflow`. Wdyt?

Reply:
Do you mean to merge the delete package with the client/mlflow? so the delete package directly call the httpCall to mlflow API?

Reply:
Yep, or even better, make the `mlflowClient` have a `artefactStorage` field. This `artefactStore` field can be of an `ArtefactStore` interface type, that's implemented by a `GCSArtefactStore` struct (which has all your implementations in the `gcs.go` file):

```go
type mlflowClient struct {
	Client        *http.Client
	Config        Config
	ArtefactStore ArtefactStore
}

type ArtefactStore interface {
	DeleteArtefact(artefactName string) error
}

type GCSArtefactStore struct {
	// ...
}

func (*GCSArtefactStore) DeleteArtefact(artefactName string) error {
	// Do some deletion...
}
```

That way, in the future if we were to replace the artefact store of the mlflow instance, we can always replace the `mlflow.ArtefactStore` with a separate implementation of `ArtefactStore` interface.

Reply:
> Does this mean that the delete package will only use the mlflow client? since the GCSArtifact interface is already there, so there's no need for the delete package to call the gcs package anymore.

There should be no delete package, per this thread. But the mlflow client will take the GCS client as dependency at initialization time:
```golang
func NewMlflowClient(httpClient *http.Client, config Config, gcspkg gcs.GcsService) *mlflowClient {
    // ...
}
```